phy/plot/features.py

```diff
@@ -0,0 +1,242 @@
+        self.n_spikes, self.n_channels, self.n_features = value.shape
+        self._features = value
+        self.set_to_bake('spikes', 'spikes_clusters', 'color')
+
+    def _check_dimension(self, dim):
+        if isinstance(dim, tuple):
+            assert len(dim) == 2
+            channel, feature = dim
+            assert 0 <= channel < self.n_channels
+            assert 0 <= feature < self.n_features
+        elif isinstance(dim, string_types):
+            assert dim == 'time'
+        else:
+            raise ValueError('{0} should be (channel, feature) '.format(dim) +
+                             'or "time".')
+
+    def _get_feature_dim(self, dim):
+        if isinstance(dim, (tuple, list)):
+            channel, feature = dim
+            return self._features[:, channel, feature]
+        elif dim == 'time':
+            return self._spike_times
+
+    def _get_mask_dim(self, dim):
+        if isinstance(dim, (tuple, list)):
+            channel, feature = dim
+            return self._masks[:, channel]
+        elif dim == 'time':
+            return np.ones(self.n_spikes)
+
+    @property
+    def dimensions(self):
+        """Dimensions."""
+        return self._dimensions
+
+    @dimensions.setter
+    def dimensions(self, value):
+        self.n_rows = len(value)
+        for dim in value:
+            self._check_dimension(dim)
+        self._dimensions = value
+        self.set_to_bake('spikes', 'spikes_clusters', 'color')
+
+    @property
+    def n_boxes(self):
+        return self.n_rows * self.n_rows
+
+    # Data baking
+    # -------------------------------------------------------------------------
+
+    def _bake_spikes(self):
+        n_points = self.n_boxes * self.n_spikes
+
+        # index increases from top to bottom, left to right
+        # same as matrix indices (i, j) starting at 0
+        positions = []
+        masks = []
+        boxes = []
+
+        for i in range(self.n_rows):
+            for j in range(self.n_rows):
+                index = self.n_rows * i + j
+
+                dim_i = self._dimensions[i]
+                dim_j = self._dimensions[j]
+                pos = np.c_[self._get_feature_dim(dim_i),
+                            self._get_feature_dim(dim_j)]
+                positions.append(pos)
+
+                # TODO: choose the mask
+                mask = self._get_mask_dim(dim_i)
+                masks.append(mask.astype(np.float32))
+                boxes.append(index * np.ones(self.n_spikes, dtype=np.float32))
+
+        positions = np.vstack(positions).astype(np.float32)
+        masks = np.hstack(masks)
+        boxes = np.hstack(boxes)
+
+        assert positions.shape == (n_points, 2)
+        assert masks.shape == (n_points,)
+        assert boxes.shape == (n_points,)
+
+        self.program['a_position'] = positions.copy()
+        self.program['a_mask'] = masks
+        self.program['a_box'] = boxes
+        self.program['u_size'] = 5.  # TODO: config
```

Comment:
the transform system is usually used to get this information, which is what those referenced lines should do... not sure if it would work for this code or not